test/runner/e2e_update_rel_test.cpp

```diff
@@ -0,0 +1,165 @@
+        "[updated property3,54]", "[950950950]"};
+    sortAndCheckTestResults(actualResult, expectedResult);
+}
+
+TEST_F(UpdateRelTest, UpdateLargeList) {
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 0 /* srcID */, 100 /* dstID */,
+                                "set e.length = 105"))
+                    ->isSuccess());
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 0 /* srcID */, 105 /* dstID */,
+                                "set e.length = 501"))
+                    ->isSuccess());
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 0 /* srcID */, 113 /* dstID */,
+                                "set e.length = 612"))
+                    ->isSuccess());
+    auto result = conn->query(
+        "MATCH (:person)-[e:knows]->(p:person) where p.ID >= 100 and p.ID <= 114 return e.length");
+    auto actualResult = TestHelper::convertResultToString(*result);
+    auto expectedResult = vector<string>{"105", "300", "303", "306", "309", "501", "315", "318",
+        "321", "324", "327", "330", "333", "612", "339"};
+    sortAndCheckTestResults(actualResult, expectedResult);
+}
+
+TEST_F(UpdateRelTest, UpdateEachElemOfSmallList) {
+    vector<string> expectedResult;
+    for (auto i = 0u; i <= 50; i++) {
+        ASSERT_TRUE(conn->query(getUpdateRelQuery("animal" /* srcTable */, "person" /* dstTable */,
+                                    "knows" /* relation */, i /* srcID */, 0 /* dstID */,
+                                    "set e.length = " + to_string(i)))
+                        ->isSuccess());
+        expectedResult.push_back(to_string(i));
+    }
+    auto result = conn->query("MATCH (:animal)-[e:knows]->(p:person) return e.length");
+    auto actualResult = TestHelper::convertResultToString(*result);
+    sortAndCheckTestResults(actualResult, expectedResult);
+}
+
+TEST_F(UpdateRelTest, UpdateEachElemOfLargeList) {
+    vector<string> expectedResult;
+    for (auto i = 1u; i <= 2500; i++) {
+        ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
+                                    "knows" /* relation */, 0 /* srcID */, i /* dstID */,
+                                    "set e.length = " + to_string(i)))
+                        ->isSuccess());
+        expectedResult.push_back(to_string(i));
+    }
+    auto result = conn->query("MATCH (:person)-[e:knows]->(p:person) return e.length");
+    auto actualResult = TestHelper::convertResultToString(*result);
+    sortAndCheckTestResults(actualResult, expectedResult);
+}
+
+TEST_F(UpdateRelTest, UpdateNewlyInsertedEdges) {
+    ASSERT_TRUE(conn->query(getInsertKnowsRelQuery(
```

Comment:
I am not sure what this tests if each of these insert queries run as a separate transaction and checkpoints. Let's discuss this.

Reply:
Added the `conn->startWriteTrx()` at the beginning of this test.